drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c

```diff
@@ -210,18 +210,26 @@ add_completion(VCHIQ_INSTANCE_T instance, VCHIQ_REASON_T reason,
 	VCHIQ_COMPLETION_DATA_T *completion;
 	DEBUG_INITIALISE(g_state.local)
 
+	mutex_lock(&instance->completion_mutex);
```

Comment:
If i get it right, the following patch reverts this patch, so this should be dropped completely.